internal.nix

```diff
@@ -62,27 +62,74 @@ rec {
   # Type: Path -> Derivation
   patchedLockfile = file: writeText "packages-lock.json" (builtins.toJSON (patchLockfile file));
 
+  # Prepared source for npm installation
+  nodeSource = nodes: runCommand "node-sources-${nodejs.version}"
```

Comment:
We might be able to get rid of this if we follow the advice from https://github.com/nodejs/node-gyp/issues/1247#issuecomment-668204686. I haven't looked into the details yet as I was happy to have something working.

Reply:
`npm_config_nodedir` is the same as passing `--nodedir=…` to npm. This is probalby as good as it gets right now.